GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -210,18 +210,14 @@ private string DescribeRevision([CanBeNull] ObjectId objectId, int maxLength = 0
         /// Provide a description for the first selected or parent to the "primary" selected last
         /// </summary>
         /// <returns>A description of the selected parent</returns>
-        private string DescribeSelectedParentRevision()
+        private string DescribeSelectedParentRevision(IEnumerable<GitRevision> parents)
         {
-            var parents = DiffFiles.SelectedItemParents
-                .Distinct()
-                .Count();
-
-            if (parents == 1)
+            if (parents.Count() == 1)
```

Comment:
We are enumerating `parents` multiple times, better to materialise the collection first.

Reply:
There are many situations where this can be optimized just in this file...